### ISSUE TITLE
Update Golang version to 1.12.x

### ISSUE THREAD

**Issue by jenspinney** [cloudfoundry-community/eirini-bosh-release#2]
The latest `eirini` source requires Go 1.12, since it relies on the new `strings.ReplaceAll()` function that was introduced in that version.

This is the error that appears during a deploy with the latest `eirini` source:
```
Task 1123 | 21:01:36 | Error: Action Failed get_task: Task 910fac1c-cb8a-4d76-7cb6-63601fb253f3 result: Compiling package eirini: Running packaging script: Running packaging script: Command exited with 2; Stdout: , Stderr: + mkdir -p /var/vcap/packages/eirini/src
+ cp -a . /var/vcap/packages/eirini/src
+ export GOPATH=/var/vcap/packages/eirini
+ GOPATH=/var/vcap/packages/eirini
++ readlink -nf /var/vcap/packages/golang
+ export GOROOT=/var/vcap/data/packages/golang/cd90ae9ceb26b29ea43c07634c8f103943d9751c
+ GOROOT=/var/vcap/data/packages/golang/cd90ae9ceb26b29ea43c07634c8f103943d9751c
+ export PATH=/var/vcap/data/packages/golang/cd90ae9ceb26b29ea43c07634c8f103943d9751c/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin
+ PATH=/var/vcap/data/packages/golang/cd90ae9ceb26b29ea43c07634c8f103943d9751c/bin:/var/vcap/bosh/bin:/usr/local/bin:/usr/local/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin
+ pushd /var/vcap/packages/eirini/src/code.cloudfoundry.org/eirini/cmd/opi
+ go build -o /var/vcap/packages/eirini/bin/opi
# code.cloudfoundry.org/eirini/k8s/utils
../../k8s/utils/names.go:10:19: undefined: strings.ReplaceAll
```

Maybe this is a good opportunity to switch to vendoring in the Golang package: https://bosh.io/docs/package-vendoring/

**Comment by viovanov** [comment]
@jenspinney the recent bump included 1.12 for go
it's still blob based though - if you have a PR to use a vendor package, I'd happily accept :)

**Comment by jenspinney** [comment]
Great, thanks for bumping that :)

When vendoring packages, you need the `final.yml` file with creds to the blobstore (vendoring takes the package from bosh.io and uploads it to the blobstore used for a given bosh release). We actually did look into submitting a PR for vendoring the go package before we realized that :)
